python/lsst/ts/Dome/mock_controller.py

```diff
@@ -143,120 +211,238 @@ async def cmd_loop(self, reader, writer):
+        )
+
+    async def crawlEl(self, **kwargs):
+        """Mock crawling the light and wind screen.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "dirMotion" with a
+            string value (UP or DOWN) and the key "elRate" with a float value.
+        """
+        self.log.info(f"Received command 'crawlEl' with arguments {kwargs}")
+        await self.lwscs.crawlEl(
+            direction=kwargs["dirMotion"], velocity=float(kwargs["elRate"])
+        )
+
+    async def setLouver(self, **kwargs):
+        """Mock setting the position of a louver.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "id" with an int
+            value and the key "position" with a float value.
+        """
+        self.log.info(f"Received command 'setLouver' with arguments {kwargs}")
+        await self.lcs.setLouver(
+            louver_id=int(kwargs["id"]), position=int(kwargs["position"])
+        )
+
+    async def closeLouvers(self):
+        """Mock closing all louvers.
+        """
+        self.log.info(f"Received command 'closeLouvers'")
+        await self.lcs.closeLouvers()
+
+    async def stopLouvers(self):
+        """Mock stopping the motion of all louvers.
+        """
+        self.log.info(f"Received command 'stopLouvers'")
+        await self.lcs.stopLouvers()
+
+    async def openShutter(self):
+        """Mock opening the shutter.
+        """
+        self.log.info(f"Received command 'openShutter'")
+        await self.apscs.openShutter()
+
+    async def closeShutter(self):
+        """Mock closing the shutter.
+        """
+        self.log.info(f"Received command 'closeShutter'")
+        await self.apscs.closeShutter()
+
+    async def stopShutter(self):
+        """Mock stopping the motion of the shutter.
+        """
+        self.log.info(f"Received command 'stopShutter'")
+        await self.apscs.stopShutter()
 
     async def config(self, **kwargs):
+        """Mock configure the lower level components.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain keys for all lower level
+            components to be configured with values that are dicts with keys for all the parameters that
+            need to be configured. The structure is
+            "AMCS":
+                "jmax"
+                "amax"
+                "vmax"
+            "LWSCS":
+                "jmax"
+                "amax"
+                "vmax"
+
+            It is assumed that the values of the configuration parameters are validated to lie within the
+            limits before being passed on to this function.
+        """
         self.log.info(f"Received command 'config' with arguments {kwargs}")
+        if kwargs["AMCS"]:
```

Comment:
I am confused what this if is supposed to do. If you want to allow AMCS itself to be optional then you can use this:
```
amcs_config = kwargs.get("AMCS")  # This returns None if absent, instead of raising
if amcs_config:
```
The rest of the if statements check if the value is non-zero (or non-None). Are they optional? Is 0 used to mean "no change"? Please document this in the doc string.
If you do want to allow optional values then you could eliminate a lot of stutter with a loop:

```
for field in ("jmax", "amax", "vmax"):
    if field in amcs_config:
        setattr(self, field, amcs_config[field])
```

Reply:
Done.